api/src/services/assets.ts

```diff
@@ -178,7 +180,20 @@ export class AssetsService {
 				readStream.unpipe(transformer);
 			});
 
-			await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			try {
+				await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			} catch (error) {
+				await storage.location(file.storage).delete(assetFilename);
+
+				if ((error as Error)?.message?.includes('timeout')) {
+					throw new ServiceUnavailableError({
+						service: 'assets',
+						reason: `Transformation timed out after ${timeoutSeconds}s`,
```

Comment:
Let's not dispose more information than necessary and remove the timeout value here.

Reply:
Makes sense! Updated in https://github.com/directus/directus/pull/19684/commits/316f4e6cc87b26d63bdb61ab1f24820ce8feb42c 👍